src/hardware/voodoo.cpp

```diff
@@ -5920,7 +5920,7 @@ static void lfb_w(uint32_t offset, uint32_t data, uint32_t mem_mask) {
 			/* make sure we care about this pixel */
 			if (mask & 0x0f)
 			{
-				int64_t iterw = sw[pix] << (30-16);
+				int64_t iterw = static_cast<int64_t>(sw[pix]) << (30-16);
```

Comment:
Another risky change; are we 100% the "fix" doesn't break anything?

Reply:
The original author used a 64-bit LHS, expecting the RHS to step up to full 64-bit range due to 14-bit left-shift.

So the code just explicitly lets that happen (and fixes a warning about it).

I'm confident that if the author really wanted to cut-off at 32-bit mark, they would have `& 0xffffffff` the result and stored it in an `int32` (because this &-mask at type-sizes is prevalent throughout the voodoo code).